inst/tests/tests.Rraw

```diff
@@ -3729,6 +3729,13 @@ test(1100, dt1[dt2,roll=-Inf,rollends=c(FALSE,TRUE)]$ind, INT(NA,NA,1,2,2,2,2,2,
   DT = data.table(x=sample(5,20,TRUE), y=sample(2,20,TRUE), z=sample(letters[1:2],20,TRUE), d1=runif(20), d2=1L)
   test(1102.38, names(dcast(DT, x ~ y + z, fun.aggregate=length, value.var = "d2", sep=".")),
                 c("x", "1.a", "1.b", "2.a", "2.b"))
+
+  # test for #5512, only compute default fill if needed.
+  DT = data.table(chr=c("a","b","b"), int=1:3)
+  test(1102.39, dcast(DT, int ~ chr, min, value.var="int"), data.table(int=1:3, a=c(1L,NA,NA), b=c(NA,2L,3L), key="int"), warning="NAs introduced by coercion to integer range")
```

Comment:
hi @MichaelChirico I resolved the conflict, but now this test is failing. 
Instead of the usual base R warning (NAs introduced by coercion to integer range) we now get a dcast-specific warning, which I guess you added in that other PR? `inf (type 'double') at RHS position 1 out-of-range(NA) or truncated (precision lost) when assigning to type 'integer' (target vector)`
I was expecting the base R warning still.
Do you think this is desirable/expected? If so I will update the test to expect the new warning.

Reply:
in detail, min is being called for every combination of int/chr, including combinations for which there are no data:
```r
> min(c())
[1] Inf
Warning message:
In min(c()) : no non-missing arguments to min; returning Inf
```
Then Inf (double) is being converted to NA_integer_ ,
```r
> as.integer(min(c()))
[1] NA
Warning messages:
1: In min(c()) : no non-missing arguments to min; returning Inf
2: NAs introduced by coercion to integer range 
```

Reply:
warning comes from here. need to think more about why:

https://github.com/Rdatatable/data.table/blob/15c127e99f8d6aab599c590d4aec346a850f1334/src/assign.c#L873

Reply:
ok I will change the test